lib/pbench/agent/results.py

```diff
@@ -194,9 +194,18 @@ def copy_result_tb(self, token: str) -> None:
                 # we got what we want before we send the request.  Also,
                 # confirm that the contents of the Content-Length header
                 # is what we expect.
-                assert "Transfer-Encoding" not in request.headers
-                assert "Content-Length" in request.headers
-                assert request.headers["Content-Length"] == str(content_length)
+                assert (
+                    "Transfer-Encoding" not in request.headers
+                ), "Upload request unexpectedly contains a `Transfer-Encoding` header"
+                assert (
+                    "Content-Length" in request.headers
+                ), "Upload request unexpectedly missing a `Content-Length` header"
+                assert request.headers["Content-Length"] == str(content_length), (
+                    "Upload request `Content-Length` header contains {} -- "
+                    "expected {}".format(
+                        request.headers["Content-Length"], content_length
+                    )
```

Comment:
I'm unable to decide whether an f-string would be easier or harder to read here. The "--" at the end of the first line looks a bit odd, and "but" (or even just ",") would probably be easier to read ... although I suspect that if we ever see the output there's not going to be any doubt what it means.

Reply:
> I'm unable to decide whether an f-string would be easier or harder to read here.

See my previous comment about Black.  (I was **amazed** at its choices for breaking and wrapping this statement!)

> The "--" at the end of the first line looks a bit odd

It only looks that way in the source code -- as you say, if the message ever appears on the screen it will look "right".

Reply:
I'll replace the `--` iff I need to make some other change.

Reply:
If it helps, I'd be fine leaving the dash-dash there.